solr/core/src/java/org/apache/solr/core/TracerConfigurator.java

```diff
@@ -46,24 +50,49 @@ public static Tracer loadTracer(SolrResourceLoader loader, PluginInfo info) {
       TracerConfigurator configurator =
           loader.newInstance(info.className, TracerConfigurator.class);
       configurator.init(info.initArgs);
+      ExecutorUtil.addThreadLocalProvider(new ContextThreadLocalProvider());
       return configurator.getTracer();
     }
     if (shouldAutoConfigOTEL()) {
       return autoConfigOTEL(loader);
     }
     if (TRACE_ID_GEN_ENABLED) {
+      ExecutorUtil.addThreadLocalProvider(new ContextThreadLocalProvider());
```

Comment:
Previously (on 9x), we would register this thing if GlobalTracer.isRegistered.  I added that.  Here on main... the situation isn't clear to me.  Wouldn't we want the equivalent?  For example, I see above shouldAutoConfigOTEL would *not* set this but don't we want it to?

Reply:
> I see above shouldAutoConfigOTEL would not set this but don't we want it to?

it does set it, see https://github.com/apache/solr/pull/1856/files/a130cc262c6a2d2244cd53e8cd23dc1b2126e60e#diff-a4b4f5abc7be3762c59d11f3919ddcadfb58bbd2d6c9346b042d8a38a2067c2aR95